components/script/dom/underlyingsourcecontainer.rs

```diff
@@ -55,6 +62,165 @@ impl UnderlyingSourceType {
     }
 }
 
+pub enum TeeCancelAlgorithm {
+    Cancel1Algorithm,
+    Cancel2Algorithm,
+}
+
+#[dom_struct]
+pub struct TeeUnderlyingSource {
+    reader: DomRoot<ReadableStreamDefaultReader>,
+    stream: Dom<ReadableStream>,
+    branch1: Option<DomRoot<ReadableStream>>,
+    branch2: Option<DomRoot<ReadableStream>>,
+    #[ignore_malloc_size_of = "Rc"]
+    reading: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    read_again: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    canceled1: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    canceled2: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    clone_for_branch2: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    reason1: Rc<Heap<*mut JSObject>>,
```

Comment:
I'm wondering this shouldn't be `Box` (see [`boxed`](https://github.com/servo/mozjs/blob/f68e385deb685e8e74f50e71d878d0889903e378/mozjs-sys/src/jsgc.rs#L314)). 

Also, I'm pretty sure it needs to be not an object but a `JSVal`.